### ISSUE TITLE
Upgrade Wasmer to 0.17.0 and  upgrade downcasting of errors

### ISSUE THREAD

**Issue by ilblackdragon** [nearprotocol/nearcore#2055]
We need to upgrade to Wasmer [0.17.0](https://github.com/wasmerio/wasmer/releases/tag/0.17.0) and upgrade downcasting of errors such that we can diffirentiate between host and guest panics, see: https://github.com/wasmerio/wasmer/issues/1328.
0.17.0 also includes fixes for the following critical bugs:
https://github.com/wasmerio/wasmer/pull/1436
https://github.com/wasmerio/wasmer/pull/1412

And it includes NaN canonicalization: https://github.com/wasmerio/wasmer/pull/1303

For more context see https://github.com/nearprotocol/nearcore/issues/2316#issuecomment-610679523 . Currently we are printing "unknown" error all the time. See the report below.

---

Receiving output in nearcore:
```
Bad error case! Output is non-deterministic TypeId { t: 7549865886324542212 } "unknown error"
```

Steps to repro:
- Deploy contract staking-contract from https://github.com/nearprotocol/stakewars/pull/110
- Create two accounts `my_validator` and `acc1`. Deposit `100` NEAR to `acc1`.
- Call `near call my_validator deposit '' --account_id acc1 --amount 10`

Output from near:
```
Error:  { Error: The used access key requires exactly one FunctionCall action
    at Object.parseRpcError (/usr/local/lib/node_modules/near-shell/node_modules/nearlib/lib/utils/rpc_errors.js:25:19)
    at Account.signAndSendTransaction (/usr/local/lib/node_modules/near-shell/node_modules/nearlib/lib/account.js:87:36)
    at process._tickCallback (internal/process/next_tick.js:68:7) type: 'ActionError', index: 0 }
```

**Comment by ilblackdragon** [comment]
Ok, isolated it to having prints in my contracts.
We should probably either block contracts with prints or make this errors more eligible.

The "used access key requires" is different errors, both in  nearlib and somewhere else. Investigating.

**Comment by ilblackdragon** [comment]
Filed the error as separate issue in nearlib - https://github.com/nearprotocol/nearlib/issues/210

**Comment by evgenykuzyakov** [comment]
The first part:
```
Bad error case! Output is non-deterministic TypeId { t: 7549865886324542212 } "unknown error"
```

Might be related to Wasm Traps. It should be addressed by https://github.com/nearprotocol/near-bindgen/pull/73

**Comment by ilblackdragon** [comment]
@evgenykuzyakov can you test and closed if it's fixed?

**Comment by nearmax** [comment]
I think it is still happening. I asked Wasmer to clarify why is it happening.

**Comment by evgenykuzyakov** [comment]
I'm not working on this

**Comment by nearmax** [comment]
Blocked by: https://github.com/wasmerio/wasmer/issues/1328

**Comment by nearmax** [comment]
Should be resolved by https://github.com/nearprotocol/nearcore/issues/2316

**Comment by fckt** [comment]
> Should be resolved by #2316

Unfortunately, Wasmer doesn't return ExceptionCode's today (for single pass backend). Wait for  wasmerio/wasmer#1328

**Comment by nearmax** [comment]
More context:
https://github.com/nearprotocol/nearcore/issues/2316#issuecomment-610679523

**Comment by nearmax** [comment]
Blocked by https://github.com/wasmerio/wasmer/issues/1338

**Comment by olonho** [comment]
Seems due to https://github.com/rust-lang/rust/issues/68905 we have to update the Rust toolchain.

**Comment by olonho** [comment]
nightly-2020-05-16 works fine.
